internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -969,12 +971,15 @@ func expandRegionConfigSpec(tfList []any, providerName string, rootDiskSizeGB *f
 		apiObject.NodeCount = conversion.Pointer(v.(int))
 	}
 
-	apiObject.DiskSizeGB = rootDiskSizeGB
-	// disk size gb defined in inner level will take precedence over root level.
 	if v, ok := tfMap["disk_size_gb"]; ok && v.(float64) != 0 {
 		apiObject.DiskSizeGB = conversion.Pointer(v.(float64))
 	}
 
+	// value defined in root is set if it is defined in the create, or value has changed in the update.
+	if rootDiskSizeGB != nil {
+		apiObject.DiskSizeGB = rootDiskSizeGB
+	}
```

Comment:
This enables changes made at the root level to be included in the update request. If disk size gb is not modified at root level only inner level is considered.